src/main/java/org/opensrp/service/PhysicalLocationService.java

```diff
@@ -485,4 +496,22 @@ public LocationTree buildLocationHierachyFromLocation(String locationId, boolean
 		return locationTree;
 	}
 
+	private void persistInRedisCache(List<AssignedLocations> assignedLocations, String username) {
+		List<Date> toDates = new ArrayList<>();
+		for(AssignedLocations assignedLocation : assignedLocations) {
+			if(assignedLocation.getToDate()!=null) {
+				toDates.add(assignedLocation.getToDate());
+			}
+		}
+		hashOps.put(username, ASSIGNED_LOCATIONS_HASH_KEY, assignedLocations);
+		redisTemplate.expire(username, findMinimumDate(toDates).getTime(), TimeUnit.MILLISECONDS);
```

Comment:
Expiry should be set only if the date is not null

Reply:
The logic also seems wrong, the seconds param is for seconds till when expiry will happen

Reply:
Updated, please review.